app/controllers/listings.controller.js

```diff
@@ -44,5 +44,39 @@ exports.createListing = function(req, res, next) {
 };
 
 exports.getById = function(req, res, next) {
+    Listing.findOne({_id: req.params.listing}, function (err, listing) {
+        if (err) {
+            res.status(500).send({successful: false, text: err.message});
+        } else if (!listing) {
+            res.status(400).send({successful: false, text: 'Can not find listing :/'});
+        } else {
+            res.json(listing);
+        }
+    }).populate('user');
+};
+
+
+exports.editListing = function (req, res, next) {
+    Listing.findOne({_id: req.body.listing}, function (err, listing) {
+        if (err) {
+            res.status(500).send({successful: false, text: err.message});
+        } else if (!listing) {
+            res.status(400).send({successful: false, text: 'Can not find listing :/'});
+        } else {
+            listing.name = req.body.name;
+            listing.description = req.body.description;
+            listing.price = req.body.price;
+            listing.imageUrl = req.body.imageUrl;
+            listing.save(function (err) {
+                if (err) {
+                    res.status(500).send({successful: false, text: err.message});
+                } else {
+                    res.status(200).send({successful: true, text: 'Edited Succesfully!'});
```

Comment:
Maybe have the response be something more descriptive such as `'Edited Listing Successfully!'` or `'Successfully edited listing!'` or maybe include the listing name, `'Successfully edited listing for ' + listing.name + '!'`.